### PR TITLE
MAISI VAE transforms

### PR DESCRIPTION
**Pull request by Can-Zhao**
Fixes # .

### Description
MAISI VAE transforms

### Checks

- [x] Avoid including large-size files in the PR.
- [x] Clean up long text outputs from code cells in the notebook.
- [x] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [x] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

**Review by KumoLiu — COMMENTED** [review]
Thanks for the pr.
It seems that this PR contains a variety of transform list dependent on mri or ct, and if it's only specific in vae training, I feel that we don't need not define classes or functions for these transforms separately. But if others don't have comments, we can do that in the first version.

File: generative/maisi/scripts/transforms.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+    compute_dtype: torch.dtype = torch.float32,
+    spacing_type: str = "original",
+    spacing: Optional[List[float]] = None,
+    image_keys: List[str] = ["image"],
+    label_keys: List[str] = [],
+    additional_keys: List[str] = [],
+    select_channel: int = 0,
+) -> tuple:
+    """
+    Define the transform pipeline for training and validation.
```
Comment (mingxin-zheng) [inline generative/maisi/scripts/transforms.py:109]: Please be more specific about the transform docstring, e.g. for variational autoencoder in MAISI

File: generative/maisi/scripts/transforms.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    interp_mode = ["bilinear"] * len(image_keys) + ["nearest"] * len(label_keys)
+
+    common_transform = [
+        SelectItemsd(keys=keys, allow_missing_keys=True),
+        LoadImaged(keys=keys, allow_missing_keys=True),
+        EnsureChannelFirstd(keys=keys, allow_missing_keys=True),
+        Orientationd(keys=keys, axcodes="RAS", allow_missing_keys=True),
+    ]
+
+    if modality == "mri":
```
Comment (mingxin-zheng) [inline generative/maisi/scripts/transforms.py:141]: Can we also check if the modality input is valid? Also I would suggest that it accepts "CT", "MRI" as valid inputs as well.

File: generative/maisi/scripts/transforms.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    """
+    Define the transform pipeline for training and validation.
+
+    Args:
+        modality (str): The imaging modality, either 'ct' or 'mri'.
+        data_aug (bool): Whether to apply random data augmentation.
+        k (int, optional): Patches should be divisible by k. Defaults to 4.
+        patch_size (List[int], optional): Size of the patches. Defaults to [128, 128, 128].
+        val_patch_size (Optional[List[int]], optional): Size of validation patches. Defaults to None.
+        compute_dtype (torch.dtype, optional): Final data type. Defaults to torch.float32.
```
Comment (mingxin-zheng) [inline generative/maisi/scripts/transforms.py:117]: I am confused about what the "final data" is when I first read this line. Does it involve multiple steps, or is it input/output? Maybe we can improve the docstring here.

File: generative/maisi/scripts/transforms.py [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+        spacing_type (str, optional): Type of spacing. Defaults to "original".
```
Comment (mingxin-zheng) [inline generative/maisi/scripts/transforms.py:118]: Please describe the acceptable values too `["original", "fixed", "rand"]:`

File: generative/maisi/scripts/transforms.py (hunk 1 above)
Comment (mingxin-zheng) [reply generative/maisi/scripts/transforms.py:109]: Also would be good to know the outcomes of different input options briefly

File: generative/maisi/scripts/transforms.py [hunk 5]
```diff
@@ -0,0 +1,292 @@
...
+    label_keys: List[str] = [],
+    additional_keys: List[str] = [],
+    select_channel: int = 0,
+) -> tuple:
+    """
+    Define the transform pipeline for training and validation.
+
+    Args:
+        modality (str): The imaging modality, either 'ct' or 'mri'.
+        data_aug (bool): Whether to apply random data augmentation.
```
Comment (mingxin-zheng) [inline generative/maisi/scripts/transforms.py:113]: Would this be better named as `random_aug`?

**Review by mingxin-zheng — APPROVED** [review]
Hi @Can-Zhao @KumoLiu , I just have some minor comments. Please feel free to either fix or skip them. Thanks!

File: generative/maisi/scripts/transforms.py (hunk 2 above)
Comment (Can-Zhao) [reply generative/maisi/scripts/transforms.py:154]: I put a warning there. It is OK to be not ct and mri, that means it uses original intensity

File: generative/maisi/scripts/transforms.py [hunk 6]
```diff
@@ -0,0 +1,292 @@
...
+        compute_dtype (torch.dtype, optional): Final data type. Defaults to torch.float32.
+        spacing_type (str, optional): Type of spacing. Defaults to "original".
+        spacing (Optional[List[float]], optional): Spacing values. Defaults to None.
+        image_keys (List[str], optional): List of image keys. Defaults to ["image"].
+        label_keys (List[str], optional): List of label keys. Defaults to [].
+        additional_keys (List[str], optional): List of additional keys. Defaults to [].
+        select_channel (int, optional): Channel to select for multi-channel MRI. Defaults to 0.
+
+    Returns:
+        tuple: A tuple containing train_transforms and val_transforms.
```
[the remark this answers is not in the document]
Comment (Can-Zhao) [reply generative/maisi/scripts/transforms.py]: Done. Thanks

File: generative/maisi/scripts/transforms.py (hunk 3 above)
Comment (Can-Zhao) [reply generative/maisi/scripts/transforms.py:126]: Output. changed docsring, thanks

File: generative/maisi/scripts/transforms.py (hunk 5 above)
Comment (Can-Zhao) [reply generative/maisi/scripts/transforms.py:122]: renamed it to rand_zoom

File: generative/maisi/scripts/transforms.py (hunk 4 above)
Comment (Can-Zhao) [reply generative/maisi/scripts/transforms.py]: Done. Thanks

File: generative/maisi/scripts/transforms.py (hunk 1 above)
Comment (Can-Zhao) [reply generative/maisi/scripts/transforms.py]: Added. Thanks

**Comment by Can-Zhao** [comment]
> Thanks for the pr. It seems that this PR contains a variety of transform list dependent on mri or ct, and if it's only specific in vae training, I feel that we don't need not define classes or functions for these transforms separately. But if others don't have comments, we can do that in the first version.

Yes, multi-modal training is used only for VAE for now. It might be used for diffusion model in the future.

File: generative/maisi/scripts/transforms.py (hunk 5 above)
Comment (mingxin-zheng) [reply generative/maisi/scripts/transforms.py:128]: Hi @Can-Zhao , sorry for the confusion. I mean should `data_aug` be renamed as `random_aug`.

File: generative/maisi/scripts/transforms.py (hunk 2 above)
Comment (mingxin-zheng) [reply generative/maisi/scripts/transforms.py:162]: Thanks!

File: generative/maisi/scripts/transforms.py (hunk 5 above)
Comment (Can-Zhao) [reply generative/maisi/scripts/transforms.py]: Done. Thanks

**Review by mingxin-zheng — APPROVED** [review]

**Pull request merged**